### DISCUSSION TITLE
Resizing window from its center

### DISCUSSION THREAD

**Discussion by michkozak** [rxhanson/RectanglePro-Community#71] [Ideas]
Hey,

Thanks to Custom Shortcuts, I'm currently trying out Rectangle as a more capable alternative to Moom.

Aside from various snapping, centering and moving windows to other displays, one of my most frequent uses of Moom was to resize windows to specific sizes via a keyboard shortcut.

This is of course achievable in Rectangle through Custom Shortcuts, but there's one thing missing: **keeping the window's center anchored at the same place**. 

In other words, when I'm resizing a window, I'd like it to shrink/grow from its center, so that its center after resizing is at the same spot on the screen as before resizing (assuming it's possible, the window is not in the corner, there's enough space, etc). There's no way to achieve that with current docking positions (Center, Top, Bottom, Left, Right, Top Left, Top Right, Bottom Left, Bottom Right, Custom).

Would be great if there was something like Anchor: Current/Window Center or something similar.

P.S. That would probably require some naming changes, as Docking Position does not play well with this concept, since it positions windows on X and Y axes — X:0 Y:0 meaning bottom left corner of the display — so DockingPosition "Current" or "Window Center" would not actually result in the desired behavior I described above.

**Comment by rxhanson** [comment]
Thanks for the explanation. This is something I have had planned for a while, but just haven't gotten around to it. I'll look to prioritize it in one of the upcoming releases.

**Reply by michkozak** [reply]
@rxhanson Hey, just wanted to ask what's the status on this. Downloaded v3 yesterday to try it out, and this behavior remains unchanged.

Not demanding or rushing, just inquiring. Any chance this will land in near future (or at all) ?

**Reply by rxhanson** [reply]
Thanks for checking back in! As you may have noticed, I haven't touched the Custom Shortcuts in quite a while, and this is still slated for when I'm back on that part of the app. Hopefully soon! My next bit of time will be focused on video tutorials and better documentation, and one of my upcoming tasks beyond that is reworking the UI for custom shortcuts & app layouts and that's where this would land.

**Reply by michkozak** [reply]
Got it! Thanks for the info.
